frontend/src/app/main/ui/ds/buttons/button.cljs

```diff
@@ -0,0 +1,31 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.ui.ds.buttons.button
+  (:require-macros
+   [app.common.data.macros :as dm]
+   [app.main.style :as stl])
+  (:require
+   [app.main.ui.ds.foundations.assets.icon :refer [icon*]]
+   [rumext.v2 :as mf]))
+
+(def button-variants (set '("primary" "secondary" "ghost" "destructive")))
+
+(mf/defc button*
+  {::mf/props :obj}
+  [{:keys [variant icon children class] :rest props}]
+  (when (some? variant) (assert (contains? button-variants variant) "Invalid variant"))
```

Comment:
I suggest put all conditions inside the assertion instead of putting assert inside a when:

```clojure
  (assert (or (not variant)
              (contains? button-variants variant))
          "expected valid variant")
```

Reply:
This is still not fixed